samples/zigbee/light_switch/src/main.c

```diff
@@ -419,15 +425,15 @@ void zboss_signal_handler(zb_bufid_t bufid)
 		ZB_ERROR_CHECK(zigbee_default_signal_handler(bufid));
 		if (status == RET_OK) {
 			/* Check the light device address */
-			if (device_ctx.bulb_params.short_addr == 0xFFFF) {
-				zb_err_code = zigbee_schedule_alarm(
-					find_light_bulb, bufid,
-					MATCH_DESC_REQ_START_DELAY);
-				ZB_ERROR_CHECK(zb_err_code);
-				zb_err_code = zigbee_schedule_alarm(
-					find_light_bulb_timeout, 0,
-					MATCH_DESC_REQ_TIMEOUT);
-				ZB_ERROR_CHECK(zb_err_code);
+			if (bulb_ctx.short_addr == 0xFFFF) {
+				bulb_ctx.bufid = bufid;
```

Comment:
Personally, I'd just free the buffer here and allocate a new one in the callback. It made sense to keep it when using ZBOSS alarms but here I just don't think we gain that much. @tomchy what do you think, is there much benefit in reusing the old buffer?

Reply:
I agree. There is much more hassle with keeping buffer references inside contexts than with adding the auto-scheduling routine.

In this application - there is no reason to keep it. It is worth reusing the buffer if:
 - The procedure has to be able to send a packet no-matter-what (a good example of that is the alarm notification or sending data requests).
 - The timing is crucial and the execution cannot wait for other contexts to free the buffer.

Reply:
@mariuszpos What is your opinion on that?

Reply:
I've changed code. Now the buffer is being cleaned. The new one is allocated in find_light_bulb().